### PR TITLE
Current http documentation status

### PR DESCRIPTION
**Pull request by jrudolph**
Extraction from #17452 with the compile errors in examples band-aided over.

A lot of review comments from #17452 still need to be addressed, I didn't get to that any more.

### CODE REVIEW

File: akka-docs-dev/rst/scala/code/docs/http/scaladsl/HttpServerExampleSpec.scala [hunk 1]
```diff
@@ -0,0 +1,257 @@
...
+
+    object Main extends App {
+      implicit val system = ActorSystem("my-system")
+      implicit val materializer = ActorFlowMaterializer()
+
+      val route =
+        path("hello") {
+          get {
+            complete {
+              <h1>Say hello to spray</h1>
```
Comment (drewhk) [inline akka-docs-dev/rst/scala/code/docs/http/scaladsl/HttpServerExampleSpec.scala]: spray?

File: akka-docs-dev/rst/scala/http/routing-dsl/directives/form-field-directives/formFields.rst [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+    def formFields(field: <FieldDef[T]>): Directive1[T]
+    def formFields(fields: <FieldDef[T_i]>*): Directive[T_0 :: ... T_i ... :: HNil]
+    def formFields(fields: <FieldDef[T_0]> :: ... <FieldDef[T_i]> ... :: HNil): Directive[T_0 :: ... T_i ... :: HNil]
+
+The signature shown is simplified and written in pseudo-syntax, the real signature uses magnets. [1]_ The type
+``<FieldDef>`` doesn't really exist but consists of the syntactic variants as shown in the description and the examples.
+
+.. [1] See `The Magnet Pattern`_ for an explanation of magnet-based overloading.
+.. _`The Magnet Pattern`: /blog/2012-12-13-the-magnet-pattern/
```
Comment (drewhk) [inline akka-docs-dev/rst/scala/http/routing-dsl/directives/form-field-directives/formFields.rst]: this is still a relative link which will now resolve against doc.akka.io

File: akka-docs-dev/rst/scala/http/routing-dsl/directives/header-directives/headerValueByType.rst [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+---------
+
+::
+
+    def headerValueByType[T <: HttpHeader: ClassTag](): Directive1[T]
+
+The signature shown is simplified, the real signature uses magnets. [1]_
+
+.. [1] See `The Magnet Pattern`_ for an explanation of magnet-based overloading.
+.. _`The Magnet Pattern`: /blog/2012-12-13-the-magnet-pattern/
```
Comment (drewhk) [inline akka-docs-dev/rst/scala/http/routing-dsl/directives/header-directives/headerValueByType.rst]: relative link

File: akka-docs-dev/rst/scala/http/routing-dsl/directives/header-directives/optionalHeaderValueByType.rst [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+Signature
+---------
+
+::
+
+    def optionalHeaderValueByType[T <: HttpHeader: ClassTag](): Directive1[Option[T]]
+
+The signature shown is simplified, the real signature uses magnets. [1]_
+
```
Comment (drewhk) [inline akka-docs-dev/rst/scala/http/routing-dsl/directives/header-directives/optionalHeaderValueByType.rst]: relative link

File: akka-docs-dev/rst/scala/http/routing-dsl/directives/header-directives/optionalHeaderValueByType.rst [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+.. [1] See `The Magnet Pattern`_ for an explanation of magnet-based overloading.
+.. _`The Magnet Pattern`: /blog/2012-12-13-the-magnet-pattern/
```
Comment (drewhk) [inline akka-docs-dev/rst/scala/http/routing-dsl/directives/header-directives/optionalHeaderValueByType.rst]: relative link. Please search for other occurences

**Comment by jrudolph** [comment]
I didn't yet change or fix the documentation, so if you reviewed it before you will now find the same things again :) There's still so much left to do that regardless from where I continue some things still won't be completely accurate.

**Comment by ktoso** [comment]
As first step to go into RC3 incomplete and not all comments addressed is fine, though the magnet pattern link and references to spray should be fixed before we merge for RC3 I think. The rest can come afterwards

**Comment by jrudolph** [comment]
Ok, will do.

**Comment by drewhk** [comment]
yes, agreed.

**Comment by drewhk** [comment]
Just the usual TLS failure...

**Pull request merged**